modules/generative-anthropic/clients/anthropic.go

```diff
@@ -0,0 +1,275 @@
+	forTask, err := a.generatePromptForTask(textProperties, task)
+	if err != nil {
+		return nil, err
+	}
+	return a.Generate(ctx, cfg, forTask)
+}
+
+func (a *anthropic) Generate(ctx context.Context, cfg moduletools.ClassConfig, prompt string) (*generativemodels.GenerateResponse, error) {
+	settings := config.NewClassSettings(cfg)
+
+	anthropicURL, err := a.getAnthropicUrl(ctx, settings.BaseURL())
+	if err != nil {
+		return nil, errors.Wrap(err, "get anthropic url")
+	}
+
+	// TODO: investigate useful of having optional system prompt as a parameter
+	input := generateInput{
+		Messages: []message{
+			{
+				Role:    "user",
+				Content: prompt,
+			},
+		},
+		Model:         settings.Model(),
+		MaxTokens:     settings.MaxTokens(),
+		StopSequences: settings.StopSequences(),
+		Temperature:   settings.Temperature(),
+		TopK:          settings.K(),
+		TopP:          settings.P(),
+	}
+
+	body, err := json.Marshal(input)
+	if err != nil {
+		return nil, errors.Wrap(err, "marshal body")
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", anthropicURL,
+		bytes.NewReader(body))
+	if err != nil {
+		return nil, errors.Wrap(err, "create POST request")
+	}
+	apiKey, err := a.getApiKey(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "Anthropic API key")
+	}
+
+	req.Header.Add("x-api-key", apiKey)
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Request-Source", "unspecified:weaviate")
+	req.Header.Add("anthropic-version", "2023-06-01")
+
+	res, err := a.httpClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "do POST request")
+	}
+
+	defer res.Body.Close()
+
+	// errors from the API are identifiable by the status code
+	if res.StatusCode != 200 {
+		bodyBytes, err := io.ReadAll(res.Body)
+		if err != nil {
+			return nil, errors.Wrap(err, "read response body")
+		}
+
+		var resBody anthropicApiError
+		if err := json.Unmarshal(bodyBytes, &resBody); err != nil {
+			return nil, errors.Wrap(err, "unmarshal response body")
+		}
+
+		return nil, fmt.Errorf("Anthropic API error: %s - %s", resBody.Error.Type, resBody.Error.Message)
+
+	}
+
+	bodyBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "read response body")
+	}
+
+	var resBody generateResponse
+	if err := json.Unmarshal(bodyBytes, &resBody); err != nil {
+		return nil, errors.Wrap(err, "unmarshal response body")
+	}
+
+	textResponse := resBody.Content[0].Text
+	if len(resBody.Content) > 0 && textResponse != "" {
+		trimmedResponse := strings.Trim(textResponse, "\n")
+		return &generativemodels.GenerateResponse{
+			Result: &trimmedResponse,
+		}, nil
+	}
+
+	return nil, errors.New("no response from Anthropic")
+}
+
+func (a *anthropic) getAnthropicUrl(ctx context.Context, baseURL string) (string, error) {
+	passedBaseURL := baseURL
+	if headerBaseURL := a.getValueFromContext(ctx, "X-Anthropic-Baseurl"); headerBaseURL != "" {
```

Comment:
please add header value to allowed header [list](https://github.com/weaviate/weaviate/blob/main/usecases/config/config_handler.go#L291)

Reply:
Didn't know this about this one, thanks!